src/subscriptions/Subscriber.ts

```diff
@@ -47,27 +47,28 @@ export default class Subscriber {
 
   /** Subscribes to subscription queries. */
   public async subscribe() {
-    Object.keys(this.subscriptionQueries).forEach(async (entity) => {
-      this.querySubscriptions[entity] = await this.graphClient.subscribe(
+    const subscriptionPromises = Object.keys(this.subscriptionQueries).map(
+      async entity => this.graphClient.subscribe(
         this.subscriptionQueries[entity],
         this.handler,
         this.fetcher,
         this.contractEntityRepository,
-      );
-    });
+      ).then((querySubscription) => {
+        Logger.info(`Subscription done for entity: ${entity}`);
+        this.querySubscriptions[entity] = querySubscription;
+      }),
+    );
+    await Promise.all(subscriptionPromises);
```

Comment:
We can try avoiding using `await` in last line of methods and return the Promise instead. This would give caller more power.

Reply:
Changes done!